webhooks/framework.go

```diff
@@ -0,0 +1,55 @@
+package webhooks
+
+import (
+	"fmt"
+	jwt "github.com/dgrijalva/jwt-go"
+	"github.com/rancher/webhook-service/drivers"
+)
+
+// ConstructPayload constructs jwt url
+func ConstructPayload(input map[string]interface{}) (map[string]interface{}, error) {
+	if _, ok := input["driver"].(string); !ok {
+		return nil, fmt.Errorf("Driver not provided")
+	}
+	driverId := input["driver"].(string)
+	driver := drivers.GetDriver(driverId)
+	if driver == nil {
+		return nil, fmt.Errorf("Driver not found")
+	}
+	payload, err := driver.ConstructPayload(input)
+	if err != nil {
+		return nil, fmt.Errorf("Error %v in driver %s for ConstructPayload\n", err, driver)
+	}
+	return payload, nil
+}
+
+//Execute does a lookup for driver and calls it
+func Execute(payload map[string]interface{}) (int, map[string]interface{}, error) {
+	jwtSigned := payload["url"]
+	token, err := jwt.Parse(jwtSigned.(string), func(token *jwt.Token) (interface{}, error) {
+		if _, ok := token.Method.(*jwt.SigningMethodRSA); !ok {
+			return nil, fmt.Errorf("Unexpected signing method: %v", token.Header["alg"])
+		}
+		return drivers.PublicKey, nil
+	})
+
+	if err != nil || !token.Valid {
+		fmt.Printf("err : %v\n", err)
+	}
```

Comment:
This gives the following error: `crypto/rsa: verification error` but still decodes with the publicKey and executes the driver to scale in/out. Verified by checking number of containers in UI. I am printing this error while I'm trying to fix it